### ISSUE TITLE
`micromamba shell -p` does not work

### ISSUE THREAD

**Issue by jonashaag** [mamba-org/mamba#2386]
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Search tried in issue tracker

-

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

Not applicable

### Describe your issue

```
$ micromamba shell -p ./myenv
critical libmamba The given prefix does not exist: "/home/j/micromamba/envs/./myenv"
```

It seems to be looking up the env in my prefix instead of by path.

cc @Hind-M 
xref #2271

**Comment by Hind-M** [comment]
Hmm I see, I think that this is because giving the `prefix` and the `name` are handled the same way in shell:
https://github.com/mamba-org/mamba/blob/main/micromamba/src/shell.cpp#L69-L72
So anything given with this option is supposed to be under `root_prefix/envs` except `base`, or when the option is not given by the user, which corresponds to `root_prefix`:
https://github.com/mamba-org/mamba/blob/main/micromamba/src/shell.cpp#L94-L102
Should we provide 2 different options, one for `name` and one for `prefix` like in the classic micromamba use?
https://github.com/mamba-org/mamba/blob/main/micromamba/src/common_options.cpp#L117-L123

**Comment by jonashaag** [comment]
Yes I think it should be consistent with eg. `micromamba create -n/-p`

**Comment by Hind-M** [comment]
cc @AntoinePrv

**Comment by jonashaag** [comment]
My experience is that shell activation (to be specific, modifying the current shell environment) is SO MUCH more trouble than just spawning a new subshell. I don't know why we have it at all, other than that Conda has it. We could get rid of almost all the shell activation code if we used `shell` only. Shell activation is especially cumbersome on Windows (Powershell and CMD) because it tends to just not work and mess up people's shells.

Personally I have stopped using shell activation almost completely and use `shell` and `run` 99% of time
